utils/swift_build_support/swift_build_support/products/swiftpm.py

```diff
@@ -10,8 +10,65 @@
 #
 # ----------------------------------------------------------------------------
 
-from . import product
+import os
+import platform
 
+from . import product
+from .. import shell
+from .. import targets
 
 class SwiftPM(product.Product):
-    pass
+    @classmethod
+    def product_source_name(cls):
+        return "swiftpm"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    def build(self, host_target):
+        run_build_script_helper('build', host_target, self, self.args)
+
+    def test(self, host_target):
+        run_build_script_helper('test', host_target, self, self.args)
+
+    def install(self, host_target):
+        run_build_script_helper('install', host_target, self, self.args)
+
+def run_build_script_helper(action, host_target, product, args):
+    script_path = os.path.join(
+        product.source_dir, 'Utilities', 'bootstrap')
+    toolchain_path = args.install_destdir
+    if platform.system() == 'Darwin':
+        # The prefix is an absolute path, so concatenate without os.path.
+        toolchain_path += \
+            product.targets.darwin_toolchain_prefix(args.install_prefix)
+
+    release = '' if args.build_variant == 'Debug' else '--release'
+
+    swiftc = os.path.join(toolchain_path, "usr", "bin", "swiftc")
```

Comment:
IIUC the idea is that you only pass the toolchain directory to SwiftPM's build script and it will find `swiftc` in there on its own. See https://github.com/apple/swift/pull/27580#pullrequestreview-300222128

Reply:
I agree but I'll do that later since SwiftPM's bootstrap script is a hot mess (which I am [reworking](https://github.com/apple/swift-package-manager/blob/master/Utilities/new-bootstrap) with the new Swift support in CMake). It'll be much easier to do this in the new script vs try to fight the old one.

Reply:
Sounds good.